src/compile/unit.ts

```diff
@@ -83,11 +83,18 @@ export class UnitModel extends Model {
 
   private _initScale(mark: Mark, encoding: Encoding, config: Config): Dict<Scale> {
     return UNIT_SCALE_CHANNELS.reduce(function(_scale, channel) {
-      if (vlEncoding.has(encoding, channel)) {
-        const scaleSpec = encoding[channel].scale || {};
-        const channelDef = encoding[channel];
+      if (vlEncoding.has(encoding, channel) ||
+          (channel === X && vlEncoding.has(encoding, X2)) ||
+          (channel === Y && vlEncoding.has(encoding, Y2))) {
+
+        let targetChannel = channel;
+        if (!vlEncoding.has(encoding, channel)) {
+          targetChannel = channel === X ? X2 : Y2;
+        }
+        const channelDef = encoding[targetChannel];
+        const scaleSpec = channelDef.scale || {};
```

Comment:
We haven't really merge the properties between `x` and `x2` if both exist yet.  
I would say just add `//TODO` for now and create a follow-up issue.  
No need to implement it yet.